src/cohort.js

```diff
@@ -0,0 +1,44 @@
+export default class Cohort {
+  constructor(name, capacity = 24) {
+    if (typeof name !== 'string') {
+      throw new Error('name input must be String')
+    }
+
+    if (name.length < 1) {
+      throw new Error('name input must not be blank')
+    }
+
+    if (typeof capacity !== 'number') {
+      throw new Error('capacity input must be Number')
+    }
+
+    this.name = name
+    this.capacity = capacity
+    this.id = Cohort.nextId()
+  }
+
+  get details() {
+    return {
+      cohortId: this.id,
+      name: this.name,
+      capacity: this.capacity
+    }
+  }
+
+  setCapacity(newCapacity) {
+    if (typeof newCapacity !== 'number') {
+      throw new Error('capacity input must be Number')
+    }
+
+    this.capacity = newCapacity
+  }
+
+  static nextId() {
+    if (!this.lastId) this.lastId = 0
+    return ++this.lastId
```

Comment:
Ive not seen 'static' before. Whats going on here?

Reply:
@PeachyOmnivore it's a function that's only visible for the class instance's method, and can't be called like a normal method – sort of a VIP function ;)

Reply:
`static` denotes a property or method on the class `Cohort` itself rather than an instance of it (`newCohort = new Cohort()`).

With this I'm applying a sequential cohort id to each new instance of the `Cohort` class I create if that makes sense?

Reply:
@Uprising5034 I see what you're doing here but let me pose you a question:

Should a blueprint care about every physical instance of a house that's created from it?

If there is ever a need for something to be "managed" in this way, surely that's the job of some kind of _manager_ class

Reply:
> Should a blueprint care about every physical instance of a house that's created from it?
@vherus No, it shouldn't be concerned about it's siblings really. I guess the idea with it being a static property was that the instances themselves were relieved from dealing with the id.

Instead should I be taking a functional approach with the blueprints? eg:
```js
export default class Cohort {
  constructor(name, id, capacity = 24) {
    ...
    this.name = name
    this.capacity = capacity
    this.id = id
    }
  ...
}
```

Reply:
@Uprising5034 absolutely. An ID is not something that an entity should manage, it's a thing that is assigned to it by the system. In this exercise, that could be the Cohort Manager; in a live app, that could be the database. An ID is just a value that's provided by something external to help identify a specific entity, so it's best to keep the entity itself dumb